tests/conformance/FirestoreTest.php

```diff
@@ -0,0 +1,368 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Tests\Conformance;
+
+use Google\Cloud\Core\Timestamp;
+use Google\Cloud\Firestore\Connection\ConnectionInterface;
+use Google\Cloud\Firestore\FieldPath;
+use Google\Cloud\Firestore\FieldValue;
+use Google\Cloud\Firestore\FirestoreClient;
+use Google\Cloud\Firestore\PathTrait;
+use Google\Cloud\Tests\ArrayHasSameValuesToken;
+use Google\GAX\Serializer;
+use GuzzleHttp\Client;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\Exception\Call\UnexpectedCallException;
+
+/**
+ * @group firestore
+ */
+class FirestoreTest extends TestCase
+{
+    use PathTrait;
+
+    const TEST_FILE = 'https://raw.githubusercontent.com/GoogleCloudPlatform/google-cloud-common/master/testing/firestore/testdata/tests.binprotos';
+
+    private $client;
+    private $connection;
+
+    private $skipped = [
+        'create: nested ServerTimestamp field', // need to strip empty maps
+        'create: multiple ServerTimestamp fields', // need to strip empty maps
+        'set: nested ServerTimestamp field', // need to strip empty maps
+        'set: multiple ServerTimestamp fields', //
+        'set-merge: Merge with a field', // need mergeFields support
+        'set-merge: Merge with a nested field', // need mergeFields support
+        'set-merge: Merge field is not a leaf', // need mergeFields support
+        'set-merge: Merge with FieldPaths', // need mergeFields support
+        'set-merge: ServerTimestamp with Merge of both fields', // need mergeFields support
+        'set-merge: If is ServerTimestamp not in Merge, no transform', // need mergeFields support
+        'set-merge: If no ordinary values in Merge, no write', // need mergeFields support
+        'update: ServerTimestamp with dotted field', // need to strip empty maps
+        'update: nested ServerTimestamp field', // need to strip empty maps
+        'update: multiple ServerTimestamp fields', // need to strip empty maps
+        'update-paths: nested ServerTimestamp field', // need to strip empty maps
+        'update-paths: multiple ServerTimestamp fields', // need to strip empty maps
+    ];
+
+    public function setUp()
+    {
+        $this->client = \Google\Cloud\Dev\stub(FirestoreClient::class, [
+            [
+                'projectId' => 'projectID'
+            ]
+        ]);
+        $this->connection = $this->prophesize(ConnectionInterface::class);
+    }
+
+    /**
+     * @dataProvider cases
+     */
+    public function testConformance($test)
+    {
+        if (in_array($test['description'], $this->skipped)) {
+            $this->markTestSkipped('manually skipped '. $test['description']);
+            return;
+        }
+
+        if (isset($test['get'])) {
+            return $this->runGet($test['get'], $test['description']);
```

Comment:
The test description doesn't appear to be used in some of the `run*` calls. Also, these chained else ifs seem like an ideal candidate for a switch statement.